tests/integration/integration_util_test.go

```diff
@@ -649,3 +650,243 @@ func testConstructFailures(t *testing.T, lang string, dependencies ...string) {
 		})
 	}
 }
+
+func testProviderConfiguration(t *testing.T, lang string) {
+	type testCase struct {
+		// Name of the test.
+		testName string
+
+		// Name of the configuration property to test.
+		prop string
+
+		// Pulumi YAML expression defining the config.
+		config any
+
+		// What the provider should receive in the request to Configure(), under GetVariables()
+		expectVariables any
+
+		// What the provider should receive in the request to Configure(), under GetArgs()
+		expectArgs any
+
+		// Which languages to limit the test to. If nil, run in every language.
+		languages func(string) bool
+	}
+
+	testCases := []testCase{
+		{
+			testName:        "s-empty",
+			prop:            "s",
+			config:          "",
+			expectVariables: map[string]any{"testconfigprovider:config:s": ""},
+			expectArgs:      map[string]any{"version": "1.0.0", "s": ""},
+			languages:       func(lang string) bool { return lang != "yaml" },
+		},
+		{
+			testName:        "s-empty-yaml",
+			prop:            "s",
+			config:          "",
+			expectVariables: map[string]any{"testconfigprovider:config:s": ""},
+			languages:       func(lang string) bool { return lang == "yaml" },
+		},
+		{
+			testName:        "s-simple",
+			prop:            "s",
+			config:          "x",
+			expectVariables: map[string]any{"testconfigprovider:config:s": "x"},
+			expectArgs:      map[string]any{"version": "1.0.0", "s": "x"},
+			languages:       func(lang string) bool { return lang != "yaml" },
+		},
+		{
+			testName:        "s-simple-yaml",
+			prop:            "s",
+			config:          "x",
+			expectVariables: map[string]any{"testconfigprovider:config:s": "x"},
+			expectArgs:      map[string]any{"s": "x"},
+			languages:       func(lang string) bool { return lang == "yaml" },
+		},
+		{
+			testName:        "s-jsonish",
+			prop:            "s",
+			config:          "{}",
+			expectVariables: map[string]any{"testconfigprovider:config:s": "{}"},
+			expectArgs:      map[string]any{"version": "1.0.0", "s": "{}"},
+			languages:       func(lang string) bool { return lang != "yaml" },
+		},
+		{
```

Comment:
This is a starter, would like to extend to other types, various forms of secreting, etc, per doc.